File: leetsync/2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.go
func maxCount(banned []int, n int, maxSum int) int {

	bannedMap := make(map[int]bool)
	for i := 0; i < len(banned); i++ {
		bannedMap[banned[i]] = true
	}
	sum, count := 0, 0
	for i := 1; i <= n; i++ {
		if _, ok := bannedMap[i]; ok {
			continue
		}
		sum += i
		if sum > maxSum {
			return count
		}
		count++
	}
	return count
}

File: leetsync/2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        Set<Integer> set = new HashSet<>();
        for (int num : banned) {
            set.add(num);
        }
        int count = 0;
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            if (set.contains(i)) {
                continue;
            }
            if ((sum + i) <= maxSum) {
                sum += i;
                count++;
            } else {
                break;
            }
        }
        return count;
    }
}